deploy/grafana-dashboard/utils/color_mapping.py

```diff
@@ -19,3 +21,9 @@ def color_mapping(name, color):
 
 
 grey_all_mapping = color_mapping("all", "#4c4b5a")
+
+color_mapping_environments = {}
+deep_update_dict(color_mapping_environments, color_mapping("p10/java:8-antlr", "yellow"))
+deep_update_dict(color_mapping_environments, color_mapping("p28/r:4", "blue"))
+deep_update_dict(color_mapping_environments, color_mapping("p29/python:3.8", "orange"))
+deep_update_dict(color_mapping_environments, color_mapping("p31/java:17", "red"))
```

Comment:
What if ... we would change the code so that there is less repetition / duplication?

My first idea was to use something like

```python3
deep_update_dict(
    color_mapping("p10/java:8-antlr", "yellow"),
    color_mapping("p28/r:4", "blue"),
    # ...
)
```

my second idea was even more reduced (too much?):

```python3
{
    "p10/java:8-antlr": "yellow",
    "p28/r:4": "blue",
    # ...
}
# and then map each key/value pair to the required `color_mapping`
```

However, as dicts in Python and using a map function is not natively supported, one could either switch to a normal array:

```python3
mapping = [
    {"p10/java:8-antlr": "yellow"},
    {"p28/r:4": "blue"}
    # ...
]

map(convert, mapping) # where convert is a function taking one argument 
```

or just stick to the first proposal which might be easier. For a variable number of arguments, have a look at args and kwargs in Python.

Reply:
Yeah, sounds good! What do you think about this `reduce` solution?

Reply:
Ah, yes. I really like that!